src/mixins/filing-template-mixin.ts

```diff
@@ -483,74 +483,71 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
    * @param entitySnapshot the latest entity snapshot
    */
   parseCorrectionFiling (filing: CorrectionFilingIF, entitySnapshot: EntitySnapshotIF = null): void {
-    // make a copy so snapshot objects are different from objects below
-    entitySnapshot = cloneDeep(entitySnapshot)
-
     // store Entity Snapshot
     this.setEntitySnapshot(entitySnapshot)
 
     // store Correction Information
-    this.setCorrectionInformation(filing.correction)
+    this.setCorrectionInformation(cloneDeep(filing.correction))
 
     // store Business Information for corrected Incorporation Application
     // *** FUTURE: change this to "if BEN correction"
     if (this.isCorrectedIncorporationApplication) {
-      this.setBusinessInformation(filing.business)
+      this.setBusinessInformation({ ...filing.business })
```

Comment:
For objects that don't have sub-objects, make a shallow copy using de-structuring.